app/views/transactions/categories/_form.html.erb

```diff
@@ -0,0 +1,38 @@
+<%= form_with model: category, data: { turbo: false } do |form| %>
+  <div class="flex flex-col space-y-4 w-96" data-controller="color-select" data-color-select-selection-value="<%= category.color %>">
+    <fieldset class="relative">
+      <span data-color-select-target="decoration" class="pointer-events-none absolute inset-y-3.5 left-3 flex items-center pl-1 block w-1 rounded-lg"></span>
+      <%= form.text_field :name,
+            value: category.name,
+            autofocus: "",
+            required: true,
+            placeholder: "Enter Category name",
+            class: "rounded-lg w-full focus:ring-black focus:border-transparent placeholder:text-gray-500 pl-6" %>
+    </fieldset>
+
+    <fieldset>
+      <%= form.hidden_field :color, data: { color_select_target: "input" } %>
+
+      <ul role="radiogroup" class="flex justify-between items-center py-2">
+        <% Transaction::Category::COLORS.each do |color| %>
+          <li tabindex="0"
+              role="radio"
+              data-action="click->color-select#select keydown.enter->color-select#select keydown.space->color-select#select"
+              data-value="<%= color %>"
+              class="flex shrink-0 justify-center items-center w-5 h-5 cursor-pointer hover:bg-gray-200 rounded-full">
+          </li>
+        <% end %>
+      </ul>
```

Comment:
The need for JS could be drastically reduced by using HTML labels with some appropriate CSS: 

```html+erb
<%= form.collection_radio_buttons :color, Transaction::Category::COLORS, :itself, :itself, item_wrapper_tag: false do |b| %>
  <%= b.label class: "has-[:checked]:bg-red-500 lex shrink-0 justify-center items-center w-5 h-5 cursor-pointer hover:bg-gray-200 rounded-full", style: "background: #{b.value};" do %> 
    <%= b.radio_button class: :hidden %>
  <% end %> 
<% end %>
```
The code above uses a hidden input nested inside its label, and make uses of the CSS `has:checked` property to conditionally style the label based on whether or not the input is checked. 

However in order to work, this uses the same background color for all checked radios. Tailwind cannot parse classes declared dynamically. 

The alternative would be to enumerate all color inputs with their hardcoded color class. 

 Those may be the reasons why the JS route was taken!

Reply:
> Those may be the reasons why the JS route was taken!

Nailed it! That's exactly the reason. I'm not convinced JS is the way to handle this either. Could feasibly declare bg-color and box-shadow TW classes for these colors and go the radio button route instead. Planning to think some more about this before marking as ready for review.

Reply:
Seems like we're using a similar approach for merchants. I think we should consolidate at some point @zachgoll. Leaving this JS implementation in for now.

Reply:
Sounds good, will keep this in mind.  I'm fine with leaving the duplicate logic here for now.

Reply:
> Could feasibly declare bg-color and box-shadow TW classes for these colors and go the radio button route instead. 

One can also use arbitrary color values with tailwind by using utiliies such as`bg-[#e99537]`. A comment can be added to the code to "trick" tailwind to generate the classes, so one can use it later on. Food for thought ! :)